lib/Dialect/AIE/Transforms/AIEObjectFifoStatefulTransform.cpp

```diff
@@ -1141,18 +1146,32 @@ struct AIEObjectFifoStatefulTransformPass
                   consumerChan.channel, 1, consumerDims);
         // generate objectFifo allocation info
         builder.setInsertionPoint(&device.getBody()->back());
+
+        // If we have PLIO then figure out the direction and make that a PLIO
+        if (producer.getPlio()) {
+          producerWireType = producer.getProducerTileOp().isShimTile()
+                                 ? WireBundle::PLIO
+                                 : WireBundle::DMA;
+          consumerWireType = !(producer.getProducerTileOp().isShimTile())
+                                 ? WireBundle::PLIO
+                                 : WireBundle::DMA;
+        } else {
+          producerWireType = WireBundle::DMA;
+          consumerWireType = WireBundle::DMA;
```

Comment:
I think this might not be needed as these are the default values assigned at variable creation.

Reply:
This if statement is in a loop that the variable creation isn't in, so they might not have the same values as the default.

Reply:
Ah right, that's my bad!